pkgs/development/libraries/glob-cpp/default.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, cpm-cmake
+, package-project-cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "glob-cpp";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "p-ranav";
+    repo = "glob";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-2y+a7YFBiYX8wbwCCWw1Cm+SFoXGB3ZxLPi/QdZhcdw=";
+  };
+
+  preConfigure = ''
+    mkdir -p ${placeholder "out"}/share/cpm
+    cp ${cpm-cmake}/share/cpm/CPM.cmake ${placeholder "out"}/share/cpm/CPM_0.27.2.cmake
+  '';
+
+  cmakeFlags = [
+    "-DCPM_SOURCE_CACHE=${placeholder "out"}/share"
```

```suggestion
    mkdir -p $out/share/cpm
    cp ${cpm-cmake}/share/cpm/CPM.cmake $out/share/cpm/CPM_0.27.2.cmake
  '';

  cmakeFlags = [
    "-DCPM_SOURCE_CACHE=$out/share"
```
can we avoid hardcoding the version number?

---

Unfortunately, we cannot. The version is hardcoded in the [source repository](https://github.com/p-ranav/glob/blob/v0.0.1/cmake/CPM.cmake).